extensions/guacamole-auth-restrict/src/main/resources/controllers/timeRestrictionFieldController.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+/**
+ * Controller for time restriction fields, which are used to select weekday and
+ * time restrictions that apply to user logins and connections.
+ */
+angular.module('guacRestrict').controller('timeRestrictionFieldController', ['$scope', '$injector',
+        function timeRestrictionFieldController($scope, $injector) {
+    
+    /**
+     * Options which dictate the behavior of the input field model, as defined
+     * by https://docs.angularjs.org/api/ng/directive/ngModelOptions
+     *
+     * @type Object.<String, String>
+     */
+    $scope.modelOptions = {
+        
+        /**
+         * Space-delimited list of events on which the model will be updated.
+         *
+         * @type String
+         */
+        updateOn : 'blur',
+
+        /**
+         * The time zone to use when reading/writing the Date object of the
+         * model.
+         *
+         * @type String
+         */
+        timezone : 'UTC'
+
+    };
+    
+    // Required types
+    // const TimeRestrictionEntry = $injector.get('TimeRestrictionEntry');
```

Comment:
?

Reply:
Okay, maybe you can help me spot what I'm doing wrong, here. I have the timeRestrictionEntry.js file, which defines this TimeRestrictionEntry - however, when I try to include it via the injector and then use the `new TimeRestrictionEntry()` constructor, it causes the fields to not be processed correctly and the following errors in the console:
```
angular.min.js:129 Failed to retrieve field with type "GUAC_TIME_RESTRICTION"
angular.min.js:129 Failed to retrieve field with type "GUAC_HOST_RESTRICTION"
angular.min.js:129 Failed to retrieve field with type "GUAC_TIME_RESTRICTION"
angular.min.js:129 Failed to retrieve field with type "GUAC_HOST_RESTRICTION"
```
I'm sure I have a typo somewhere or am doing something wrong that's really simple, but I cannot figure out what it is.

Reply:
Yeah, I'm still stumped by this. If I try to add this line into the `timeRestrictionFieldController.js` file, the field does not work at all:
```
const TimeRestrictionEntry = $injector.get('TimeRestrictionEntry');
```

At present, I've just defined the `TimeRestrictionEntry` and `HostRestrictionEntry` "types" in the controller files, which works fine since they aren't actually used outside of those files. Any insight on why the `factory()` method doesn't seem to be working as expected to create these data types would be appreciated.

Reply:
Well it looks like your type is called `timeRestrictionEntry` in [the factory](https://github.com/apache/guacamole-client/pull/830/files#diff-e1c9d1ad21ea5614d05826fa67243d11c4808c1c28d7b3ca166b583c2b4bac19R23) (note the case difference).

Reply:
Yep, that was it. Should be good, now.